probabilistic.go

```diff
@@ -0,0 +1,1142 @@
+	CFReserve(ctx context.Context, key string, capacity int64) *StatusCmd
+	//TODO LoadChunk and ScanDump missing
+
+	CMSIncrBy(ctx context.Context, key string, elements ...interface{}) *IntSliceCmd
+	CMSInfo(ctx context.Context, key string) *CMSInfoCmd
+	CMSInitByDim(ctx context.Context, key string, width, height int64) *StatusCmd
+	CMSInitByProb(ctx context.Context, key string, errorRate, probability float64) *StatusCmd
+	CMSMerge(ctx context.Context, destKey string, sourceKeys ...string) *StatusCmd
+	CMSMergeWithWeight(ctx context.Context, destKey string, sourceKeys map[string]int) *StatusCmd
+	CMSQuery(ctx context.Context, key string, elements ...interface{}) *IntSliceCmd
+
+	TOPKAdd(ctx context.Context, key string, elements ...interface{}) *StringSliceCmd
+	TOPKCount(ctx context.Context, key string, elements ...interface{}) *IntSliceCmd
+	TOPKIncrBy(ctx context.Context, key string, elements ...interface{}) *StringSliceCmd
+	TOPKInfo(ctx context.Context, key string) *TOPKInfoCmd
+	TOPKList(ctx context.Context, key string) *StringSliceCmd
+	TOPKListWithCount(ctx context.Context, key string) *MapStringIntCmd
+	TOPKQuery(ctx context.Context, key string, elements ...interface{}) *BoolSliceCmd
+	TOPKReserve(ctx context.Context, key string, k int) *StatusCmd
+	TOPKReserveWithOptions(ctx context.Context, key string, k int, width, depth int64, decay float64) *StatusCmd
+
+	TDigestAdd(ctx context.Context, key string, elements ...float64) *StatusCmd
+	TDigestByRank(ctx context.Context, key string, rank ...uint) *FloatSliceCmd
+	TDigestByRevRank(ctx context.Context, key string, rank ...uint) *FloatSliceCmd
+	TDigestCDF(ctx context.Context, key string, elements ...float64) *FloatSliceCmd
+	TDigestCreate(ctx context.Context, key string) *StatusCmd
+	TDigestCreateWithCompression(ctx context.Context, key string, compression int) *StatusCmd
+	TDigestInfo(ctx context.Context, key string) *TDigestInfoCmd
+	TDigestMax(ctx context.Context, key string) *FloatCmd
+	TDigestMin(ctx context.Context, key string) *FloatCmd
+	TDigestQuantile(ctx context.Context, key string, elements ...float64) *FloatSliceCmd
+	TDigestRank(ctx context.Context, key string, values ...float64) *IntSliceCmd
+	TDigestReset(ctx context.Context, key string) *StatusCmd
+	TDigestRevRank(ctx context.Context, key string, values ...float64) *IntSliceCmd
+	TDigestTrimmedMean(ctx context.Context, key string, lowCutQuantile, highCutQuantile float64) *FloatCmd
+}
+
+type BFInsertOptions struct {
+	Capacity   int64
+	Error      float64
+	Expansion  int64
+	NonScaling bool
+	NoCreate   bool
+}
+
+type BFReserveOptions struct {
+	Capacity   int64
+	Error      float64
+	Expansion  int64
+	NonScaling bool
+}
+
+type CFReserveOptions struct {
+	Capacity      int64
+	BucketSize    int64
+	MaxIterations int64
+	Expansion     int64
+}
+
+type CFInsertOptions struct {
+	Capacity int64
+	NoCreate bool
+}
+
+type BFInfoArgs int
+
+const (
+	BFCAPACITY BFInfoArgs = iota
+	BFSIZE
+	BFFILTERS
+	BFITEMS
+	BFEXPANSION
+)
```

Comment:
What about replacing these constants with methods (and removing constants):

```
BFInfoCapacity()
BFInfoSize()
...
BFInfoExpansion()
```

It is the same amount of work / API surface, but methods are more readable / easier to use.

Reply:
The more I look at this, the more I think Nic did a nice thing with the constants because the IDE makes the needed arguments very easily discoverable.
If we keep a `BFInfoArgs` type with `Capacity()`, `Size()` etc methods, we would have to instantiate an empty object in code, when calling `client.BFInfoArg`. Not terrible, but also not the prettiest. And since the parameter then would be a string, the users might not know we provide the helper methods above and might pass in a hard-coded string.
Does this sound about right @vmihailenco , or I missed your point?

Reply:
Instead of this:

```go
rdb.BFInfoArg(ctx, key, redis.BFCAPACITY)
```

User will write this:

```go
rdb.BFInfoCapactiy(ctx, key)
```

The implementation will look like this:

```go
func (c) BFInfoCapacity(ctx context.Context, key string) {
    return c.bfInfo(ctx, key, "capacity")
}
```
Internally, we can keep the constants and `BFInfoArg` method, but unexport them so users can't (mis)use them.

Reply:
Ahh ok! So I had misunderstood you before :) 
Let me work in it...

Reply:
Done!